src/native/managed/cdacreader/tests/TargetTests.cs

```diff
@@ -0,0 +1,348 @@
+    public void ReadIndirectGlobalValue(bool isLittleEndian, bool is64Bit)
+    {
+        int pointerSize = is64Bit ? sizeof(ulong) : sizeof(uint);
+        Span<byte> pointerData = stackalloc byte[TestGlobals.Length * pointerSize];
+        for (int i = 0; i < TestGlobals.Length; i++)
+        {
+            var (_, value, _) = TestGlobals[i];
+            WritePointer(pointerData.Slice(i * pointerSize), value, isLittleEndian, pointerSize);
+        }
+
+        string globalsJson = string.Join(',', TestGlobals.Select((g, i) => $"\"{g.Name}\": {(g.Type is null ? $"[{i}]" : $"[[{i}], \"{g.Type}\"]")}"));
+        byte[] json = Encoding.UTF8.GetBytes($$"""
+        {
+            "version": 0,
+            "baseline": "empty",
+            "contracts": {},
+            "types": {},
+            "globals": { {{globalsJson}} }
+        }
+        """);
+        Span<byte> descriptor = stackalloc byte[ContractDescriptor.Size(is64Bit)];
+        ContractDescriptor.Fill(descriptor, isLittleEndian, is64Bit, json.Length, pointerData.Length / pointerSize);
+        fixed (byte* jsonPtr = json)
+        {
+            ReadContext context = new ReadContext
+            {
+                ContractDescriptor = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(descriptor)),
+                ContractDescriptorLength = descriptor.Length,
+                JsonDescriptor = jsonPtr,
+                JsonDescriptorLength = json.Length,
+                PointerData = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(pointerData)),
+                PointerDataLength = pointerData.Length
+            };
+
+            bool success = Target.TryCreate(ContractDescriptorAddr, &ReadFromTarget, &context, out Target? target);
+            Assert.True(success);
+
+            // Indirect values are pointer-sized, so max 32-bits for a 32-bit target
+            var expected = is64Bit
+                ? TestGlobals
+                : TestGlobals.Select(g => (g.Name, g.Value & 0xffffffff, g.Type)).ToArray();
+
+            ValidateGlobals(target, expected);
+        }
+    }
+
+    private static void WritePointer(Span<byte> dest, ulong value, bool isLittleEndian, int pointerSize)
+    {
+        if (pointerSize == sizeof(ulong))
+        {
+            if (isLittleEndian)
+            {
+                BinaryPrimitives.WriteUInt64LittleEndian(dest, value);
+            }
+            else
+            {
+                BinaryPrimitives.WriteUInt64BigEndian(dest, value);
+            }
+        }
+        else if (pointerSize == sizeof(uint))
+        {
+            if (isLittleEndian)
+            {
+                BinaryPrimitives.WriteUInt32LittleEndian(dest, (uint)value);
+            }
+            else
+            {
+                BinaryPrimitives.WriteUInt32BigEndian(dest, (uint)value);
+            }
+        }
+    }
+
+    private static void ValidateGlobals(Target target, (string Name, ulong Value, string? Type)[] globals)
```

Comment:
I usually make these kind of validation methods that some `[CallerMemberName]`, `[CallerFilePath]`, `[CallerLineNumber]` arguments and use them in the "message" argument of `Assert.Equal(expected, actual, message)` so that it's straightforward to go to the failure just from the exception message instead of having to look at the stack trace